src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -238,9 +238,6 @@ static function (array $a, array $b): int {
         $pool->replaceArgument(1, $admins);
         $pool->replaceArgument(2, $groups);
         $pool->replaceArgument(3, $classes);
-
-        $routeLoader = $container->getDefinition('sonata.admin.route_loader');
-        $routeLoader->replaceArgument(1, $admins);
```

Comment:
These `$admins` are the same that the ones injected to the `Pool` (5 lines above).